File: src/main/java/com/tencent/nft/entity/BaseEntity.java
package com.tencent.nft.entity;

import java.time.LocalDateTime;

/**
 * @author: imhuis
 * @date: 2021/10/1
 * @description:
 */
public class BaseEntity {

    private Long id;
    private LocalDateTime createTime = LocalDateTime.now();
    private LocalDateTime updateTime = LocalDateTime.now();

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDateTime getCreateTime() {
        return createTime;
    }

    public void setCreateTime(LocalDateTime createTime) {
        this.createTime = createTime;
    }

    public LocalDateTime getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(LocalDateTime updateTime) {
        this.updateTime = updateTime;
    }
}

File: src/main/java/com/tencent/nft/entity/BaseEntity.kt
package com.tencent.nft.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreType
import com.fasterxml.jackson.annotation.JsonView
import java.io.Serializable
import java.time.LocalDateTime

/**
 * @author: imhuis
 * @date: 2020/1/28
 * @description:
 */
open class BaseEntity : Serializable {

    interface BaseViewGroup

    var id: Long? = null

//    @CreatedDate
    @JsonIgnore
    @JsonView(BaseViewGroup::class)
    var createTime: LocalDateTime? = LocalDateTime.now()

//    @LastModifiedDate
    @JsonIgnore
    @JsonView(BaseViewGroup::class)
    var updateTime: LocalDateTime? = LocalDateTime.now()
}